### ISSUE TITLE
How to implement filtering for each column in ngx swimlane ?

### ISSUE THREAD

**Issue by sagar-sk15** [swimlane/ngx-datatable#1445]
I want to have multi-select drop down under table headers which contains the Distinct values within that column as shown below Image:

![image](https://user-images.githubusercontent.com/30797323/41096641-dec1ec4c-6a72-11e8-965e-fab7cc909b2e.png)

**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, post on Stackoverflow or Gitter
```

**Current behavior**
Currently I have added the multiselect control under behavior, but i am not able to select any values in that multiselect dropdown. It is acting like a disable control

**Expected behavior**
It should select values in dropdown control and accordingly filter the data in grid

**Reproduction of the problem**

**What is the motivation / use case for changing the behavior?**

**Please tell us about your environment:**

* **Table version:** 0.8.x

* **Angular version:** 2.0.x

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

**Comment by hosseinGanjyar** [comment]
Did you find any solution?

**Comment by sagarsk15** [comment]
@hosseinGanjyar Nope, I have replaced multi-select drop- down with normal drop-down and its working in Chrome as of now (not in edge browser)

**Comment by hosseinGanjyar** [comment]
If using of textbox, please read this issue:
https://github.com/swimlane/ngx-datatable/issues/1455

**Comment by adarsha17** [comment]
Hi Sagar, I believe you are using ngx-datatable. Could you please share this filtering code with me. I have a similar requirement for applying filters in the table

**Comment by sagar-sk15** [comment]
@adarsha17 I have achieved this functionality by using text-box control.  As i stated in the problem, if i use drop-down control then i was not able to select the values in it, so as a work around i have used textboxes where can type free text for filtering.

**Comment by adarsha17** [comment]
Thanks Sagar! so you mean to say you used only a single text box(placed above the grid)that would accept any free text and would filter on any of the columns?

**Comment by sagar-sk15** [comment]
@adarsha17 No, what i mean is. I have kept radio button to let user select if he wants multifilter or single filter, if user selectes multifilter then text-boxes will appear below the each header of the grid. PFB
![image](https://user-images.githubusercontent.com/30797323/54415398-2bc06500-4722-11e9-82df-508ac71a46ad.png)

**Comment by adarsha17** [comment]
Thanks sagar for the clarification. Is it possible could you share the sample code?

**Comment by hirenkukadiya** [comment]
```
<ngx-datatable-column name="Task" [width]="180"> 
 <ng-template let-column="column" ngx-datatable-header-template>
    {{column.name}}
  <input type="text" class="managetextfilter" placeholder="Type to filter"(keyup)="updateFilter($event,'name')"/> 
 </ng-template>
  <ng-template ngx-datatable-cell-template let-rowIndex="rowIndex" let-value="value" let-row="rows">
    <span title="Double click to edit"  (dblclick)="editing[rowIndex + '-task'] = true" *ngIf="!editing[rowIndex + '-task']">{{value}}</span>
  </ng-template>
</ngx-datatable-column>
```

**Comment by grantwforsythe** [comment]
I took a _reactive_ approach on implementing individual column filters. You can see a demo [here](https://stackblitz.com/edit/stackblitz-starters-l6wabw5t?file=src%2Fmain.ts).
